### PR TITLE
disable update existing jira issues with parameter

### PR DESCRIPTION
**Pull request by dvag-holger-waschke**
add 3 parameter to disable update of summary, description and reopening of tickets. its no braking change if the parameter is not set it defaults to true and update of summary, description and reopen will work as before.

```
[Unit]
Description=Prometheus Jralert Alerting
Documentation=https://github.com/prometheus-community/jiralert
Wants=network-online.target
After=network.target network-online.target

[Service]
Type=simple
User=alertmanager
Group=alertmanager
ExecReload=/bin/kill -HUP $MAINPID
ExecStart=/usr/local/bin/jiralert_1.2 \
  -config /etc/jiralert/config.yml \
  -listen-address 0.0.0.0:9098 \
  -hash-jira-label \
  -update-description=false \
  -update-summary=false \
  -reopen-tickets=false
```

### CODE REVIEW

File: cmd/jiralert/main.go [hunk 1]
```diff
@@ -48,6 +48,7 @@ var (
 	logFormat     = flag.String("log.format", logFormatLogfmt, "Log format to use ("+logFormatLogfmt+", "+logFormatJSON+")")
 	hashJiraLabel = flag.Bool("hash-jira-label", false, "if enabled: renames ALERT{...} to JIRALERT{...}; also hashes the key-value pairs inside of JIRALERT{...} in the created jira issue labels"+
 		"- this ensures that the label text does not overflow the allowed length in jira (255)")
+	updateDisabled = flag.Bool("updateDisabled", false, "if enabled: updates the summary and description of the existing jira issue")
```
Comment (bwplotka) [inline cmd/jiralert/main.go:51]:
```suggestion
	updateContentDisabled = flag.Bool("update-content-disabled", false, "When true, jiralert does not update the summary and description of the existing jira issue, even when changes are spotted.")
```

File: pkg/notify/notify.go [hunk 2]
```diff
@@ -87,17 +87,19 @@ func (r *Receiver) Notify(data *alertmanager.Data, hashJiraLabel bool) (bool, er
 
 	if issue != nil {
```
Comment (bwplotka) [inline pkg/notify/notify.go:88]:
```suggestion
	if issue != nil && !updateDisabled {
```

**Review by bwplotka — CHANGES_REQUESTED** [review]
Thanks! Some nits and good to go!

Curious, what exactly is causing you to want to stop this behaviour? 🤗

**Comment by dvag-holger-waschke** [comment]
> Thanks! Some nits and good to go!
> 
> Curious, what exactly is causing you to want to stop this behaviour? 🤗

Well we have 24/7 team to monitor jira issues and they say they get confused if old issues are being updated which they might have out of sight already. as they are 'customers' for us we deliver them their wishes ;-)

I made some adjustments but they could be to specific for the main branch:

Update function only on the summary of the issue but not on the description -> you see if the alert is resolved at first sight but they dont want constant updates, as we have per example a value as label which is constant changing causing email notifcation flodding.
So I would suggest we leave the changes here as it is and I will keep the fork.

**Comment by kotarf** [comment]
Is this PR making it into a release any time soon? The current behavior of updating existing tickets renders jiralert completely non-usable by our organization. Thanks.

**Comment by dvag-holger-waschke** [comment]
I fixed a bug and made the names of the booleans more clear (remove the double negation)

Systemd unit would look like this. If the parameter is not set default is true and the update function works, so its no breaking change. we running this version in production without problems

```bash
[Unit]
Description=Prometheus Jralert Alerting
Documentation=https://github.com/prometheus-community/jiralert
Wants=network-online.target
After=network.target network-online.target

[Service]
Type=simple
User=alertmanager
Group=alertmanager
ExecReload=/bin/kill -HUP $MAINPID
ExecStart=/usr/local/bin/jiralert_1.2 \
  -config /etc/jiralert/config.yml \
  -listen-address 0.0.0.0:9098 \
  -hash-jira-label \
  -update-description=false \
  -update-summary=false \
  -reopen-tickets=false
```

**Pull request merged**